### ISSUE TITLE
input permitting multiple files

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#5789]
We currently use the rio::import to import a single file and rio::import_list to import multiple Excel sheets from an Excel book.
We quite often get multiple files to import, where there is a single file for each climatic station or for each element, or both.  Currently we have to import these files one-at a time.
This extension is to be able to import multiple files.
a) In the browse we simply choose more than one file.  I assume they would all have the same extension and we may wish to check this?
b) In the dialogue, this could result in a similar display we currently have for an Excel book.  But we don't need all the selection - we did that when we chose the files.  So we might have a listview control to see the list of the files.
c) Unless it is easy we would not have a preview.  If it is easy, then we might show a preview of the first file.
There are all sorts of ways we could spend a lot of time and make this complicated.  Let's keep it as simple as possible!

**Comment by Patowhiz** [comment]
I can work on this once PR #5738 is merged.

**Comment by rdstern** [comment]
Great - thanks @Patowhiz

**Comment by rdstern** [comment]
This issue has come back again, with problems on data input from Zambia.  Could we change the browse so you can select multiple files?  They then appear as do Excel sheets?
Almost all the situations for me will be where the file extension is the same.  So that could perhaps be checked.  Maybe choose all with the same extension as the first file.  They will come into separate sheets, so should there also be a limit on the number?

I think we need to select the file names, because we need them to be sheet names?  So just selecting a directory would not be ideal?

**Comment by Patowhiz** [comment]
@rdstern and what should happen if a file is an excel file with multiple sheets in it?

**Comment by Patowhiz** [comment]
We may need to have  a wider discussions on how to well handle different scenarios consistently with user expectations.  A simpler way could be first implementing this feature with clear limitations.

**Comment by rdstern** [comment]
I agree to keep it simple.  My first impression on your previous question is that it does not apply to multiple Excel files, because they are already potentially multiple already.  Possibly it also doesn't apply to multiple rds files, because they can also provide multiple data frames.

Maybe it should be renamed to "input permitting multiple data frames"

**Comment by Patowhiz** [comment]
@rdstern yes, and that means probably running vb.net and R code in the background to validate the permitted files, in the case of checking files that have multiple sheets in them.

@dannyparsons what's your suggestion on this?

**Comment by dannyparsons** [comment]
Further improvements for later:

- Can we have the options still when we have multiple files? They can be passed to `rio::import_list` in the same way so should be easy.
- I don't think we need to pass the `data_names` parameter to `import_data` because `files <- rio::import_list` should be a named list containing the file names already. It doesn't matter too much but if there were lots and lots of files then the code would be very long if we had `data_names` for all files. So this would be a cleaner option.

**Comment by Patowhiz** [comment]
@dannyparsons would you like me do the item (1) in this PR? I have done item (2)
